src/Storages/Freeze.cpp

```diff
@@ -0,0 +1,201 @@
+#include <Storages/Freeze.h>
+
+#include <Disks/ObjectStorages/IMetadataStorage.h>
+#include <Storages/PartitionCommands.h>
+#include <Common/escapeForFileName.h>
+#include <Common/logger_useful.h>
+
+namespace DB
+{
+void FreezeMetaData::fill(const StorageReplicatedMergeTree & storage)
+{
+    is_replicated = storage.supportsReplication();
+    is_remote = storage.isRemote();
+    replica_name = storage.getReplicaName();
+    zookeeper_name = storage.getZooKeeperName();
+    table_shared_id = storage.getTableSharedID();
+}
+
+void FreezeMetaData::save(DiskPtr data_disk, const String & path) const
+{
+    auto metadata_storage = data_disk->getMetadataStorage();
+
+    auto file_path = getFileName(path);
+    auto tx = metadata_storage->createTransaction();
+    WriteBufferFromOwnString buffer;
+
+    writeIntText(version, buffer);
+    buffer.write("\n", 1);
+    writeBoolText(is_replicated, buffer);
+    buffer.write("\n", 1);
+    writeBoolText(is_remote, buffer);
+    buffer.write("\n", 1);
+    writeString(replica_name, buffer);
```

Comment:
What will happen if I create a replica with weird name? (I thought it's not possible, but it is)
```
dell9510 :) create table kek (n int) engine=ReplicatedMergeTree('/test/kekpatherqgac/', 'r
            1') order by tuple()

CREATE TABLE kek
(
    `n` int
)
ENGINE = ReplicatedMergeTree('/test/kekpatherqgac/', 'r\n1')
ORDER BY tuple()

Ok.

0 rows in set. Elapsed: 0.032 sec. 
dell9510 :) select replica_name from system.replicas where table='kek'

SELECT replica_name
FROM system.replicas
WHERE table = 'kek'

Query id: f8422135-3d4a-4d2a-845e-6be49c4c1e55

┌─replica_name─┐
│ r
1           │
└──────────────┘

1 row in set. Elapsed: 0.019 sec. 
```